src/wallet/rpcwallet.cpp

```diff
@@ -1053,27 +1071,39 @@ static UniValue ListReceived(const CWallet& wallet, const UniValue& params, bool
 
     bool has_filtered_address = false;
     CTxDestination filtered_address = CNoDestination();
-    if (!by_label && params.size() > 3) {
+    if (!by_label && !params[3].isNull() && !params[3].get_str().empty()) {
         if (!IsValidDestinationString(params[3].get_str())) {
             throw JSONRPCError(RPC_WALLET_ERROR, "address_filter parameter was invalid");
         }
         filtered_address = DecodeDestination(params[3].get_str());
         has_filtered_address = true;
     }
 
+    // getreceivedbyaddress has address filter at 3rd index, so next params index is 4
+    // getreceivedbylabel has no filter parameter, so next params index is 3
+    int next_index{by_label ? 3 : 4};
+    bool include_immature{!params[next_index].isNull() && params[next_index].get_bool()};
+
+    // Filtering out coinbase outputs is deprecated
+    // It can be enabled by setting deprecatedrpc=filter_coinbase
+    bool filter_coinbase{pwallet->chain().rpcEnableDeprecated("filter_coinbase")};
```

Comment:
getting a compile error after the latest rebase: 
```
wallet/rpcwallet.cpp:1089:26: error: use of undeclared identifier 'pwallet'
    bool filter_coinbase{pwallet->chain().rpcEnableDeprecated("filter_coinbase")};
```

```suggestion
    bool filter_coinbase{wallet->chain().rpcEnableDeprecated("filter_coinbase")};
```

Reply:
Oof forgot to include that with the rebase. Thanks for checking!